### PR TITLE
fix: set dry_run to be optional.

### LINKED ISSUE #35166: dry_run not optional in api for set task instances state
### Apache Airflow version

2.7.2

### What happened

Sent request without dry_run parameter to dags/{dag_id}/updateTaskInstancesState and got a 500 error.

### What you think should happen instead

I should be able to send a request to update task instances state and get a valid response.

### How to reproduce

You can see this by commenting out line 215 in tests/api_connexion/schemas/test_task_instance_schema.py and running tests. This is a similar error to #34563.

### Operating System

Ubuntu 20.04

### Versions of Apache Airflow Providers

_No response_

### Deployment

Docker-Compose

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by austinau**
Fixes #35166. Duplicates the solution in #34568.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]

File: tests/api_connexion/schemas/test_task_instance_schema.py [hunk 1]
```diff
@@ -236,6 +236,22 @@ def test_success(self):
...
+            "execution_date": dt.datetime(2020, 1, 1, 0, 0, tzinfo=dt.timezone(dt.timedelta(0), "+0000")),
+            "include_downstream": True,
+            "include_future": True,
+            "include_past": True,
+            "include_upstream": True,
+            "new_state": "failed",
+            "task_id": "print_the_context",
+        }
+        assert expected_result == result
+
```
Comment (Taragolis) [inline tests/api_connexion/schemas/test_task_instance_schema.py]: Extra spaces here, it should be removed for pass [Static Checks](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#static-code-checks)

Comment (utkarsharma2) [reply tests/api_connexion/schemas/test_task_instance_schema.py]: May want to setup pre-commit hooks - https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst#id12

Comment (eladkal) [inline tests/api_connexion/schemas/test_task_instance_schema.py]:
```suggestion

```

**Comment by Calder-Ty** [comment]
Any update on this PR?

**Review by eladkal — APPROVED** [review]

**Pull request merged**